BaragonAgentService/src/main/java/com/hubspot/baragon/agent/managers/AgentRequestManager.java

```diff
@@ -177,9 +193,15 @@ public Response processRequest(String requestId,
           return reload(request, delayReload);
         case GET_RENDERED_CONFIG:
           return getRenderedConfigs(request.getLoadBalancerService().getServiceId());
+        case PURGE_CACHE:
+          return purgeCache(request.getLoadBalancerService().getServiceId());
         case REVERT:
           serviceId = request.getLoadBalancerService().getServiceId();
           return revert(request, maybeOldService, existingUpstreams.computeIfAbsent(serviceId, (key) -> new ArrayList<>()), delayReload, batchItemNumber);
+        case UPDATE_AND_PURGE_CACHE:
+          serviceId = request.getLoadBalancerService().getServiceId();
+          purgeCache(maybeOldService.or(request.getLoadBalancerService()).getServiceId());
```

Comment:
wouldn't we want to purge cache after the apply so that we get the newer content?